library/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -132,80 +220,249 @@
+                # use "/" as path
+                path += "/"
+
+            result = idrac.update_mgr.update_from_repo_url(
+                ipaddress=p.netloc, share_type=p.scheme, share_name=path,
+                share_user=share_user, share_pwd=share_pwd,
+                catalog_file=catalog_file_name, apply_update=apply_update,
+                reboot_needed=reboot, ignore_cert_warning=ignore_cert_warning,
+                job_wait=job_wait)
+
+    return result
+
+
+def update_firmware_from_net_share(idrac, share_name, share_user, share_pwd,
+                                   share_mnt, catalog_file_name,
+                                   apply_update=True, reboot=False,
+                                   job_wait=True):
+    """
+    Update firmware from a repository on a remote network share (CIFS, NFS)
+
+    :param idrac: iDRAC connection object
+    :type idrac: iDRAC connection object
+
+    :param share_name: CIFS or NFS network share
+    :type share_name: ``str``
+
+    :param share_user: username for the remote network share
+    :type share_user: ``str``
+
+    :param share_pwd: password for the remote network share
+    :type share_pwd: ``str``
+
+    :param share_mnt: local mount point for the remote network share
+    :type share_mnt: ``str``
+
+    :param catalog_file_name: Name of Catalog file on the repository
+    :type catalog_file_name: ``str``
+
+    :param apply_update: Apply update
+    :type apply_update: ``bool``
+
+    :param reboot: Reboot system
+    :type reboot: ``bool``
+
+    :param job_wait: Wait for JOB to be completed
+    :type job_wait: ``bool``
+
+    :returns: A dict containing the return value for the Update Job
+    :rtype: ``dict``
+    """
+
+    net_share_repo = FileOnShare(remote=share_name,
+                                 mount_point=share_mnt,
+                                 creds=UserCredentials(share_user, share_pwd),
+                                 isFolder=True)
+    catalog_path = net_share_repo.new_file(catalog_file_name)
+
+    result = idrac.update_mgr.update_from_repo(catalog_path=catalog_path,
+                                               apply_update=apply_update,
+                                               reboot_needed=reboot,
+                                               job_wait=job_wait)
+
+    return result
 
 
 def update_firmware(idrac, module):
-    """Update firmware from a network share and return the job details."""
-    msg = {}
-    msg['changed'] = False
-    msg['update_status'] = {}
+    """
+    Update firmware from a repository (CIFS, NFS, HTTP, HTTPS, FTP)
+
+    :param idrac: iDRAC handle
+    :type idrac: ``iDRAC Connection Object``
+
+    :param module: Ansible module
+    :type module: ``class AnsibleModule``
+
+    :returns: A tuple containing a result dict and error flag
+    :rtype: ``tuple``
+    """
+
+    result = {}
+    result['changed'] = False
+    result['failed'] = False
```

Comment:
Where are you using this variable `result['failed'] = False`

Reply:
Since module.fail_json automatically sets the failed flag, I guess it can be removed.